### DISCUSSION TITLE
testing spatial relation - Standard for Geographic information

### DISCUSSION THREAD

**Discussion by ElCep** [cormas/cormas#779] [Ideas]
The OGC has written standards for the implementation of space operators. Following "[OpenGIS ® Implementation Standard for Geographic information - Simple feature access](https://www.ogc.org/standards/sfa)" it will be great to have in CORMAS different type of object as agents but also usable by agents (p. 14 uml): 
* polygones 
* points
* lines

And different type methods for testing spatial relation (6.1.2.3 p.  16): 
* Equals : Returns 1 (TRUE) if this geometric object is “spatially equal” to anotherGeometry
* Disjoint : Returns 1 (TRUE) if this geometric object “spatially intersects” anotherGeometry.
* Touches : Returns 1 (TRUE) if this geometric object “spatially touches” anotherGeometry.
* Crosses : Returns 1 (TRUE) if this geometric object “spatially crosses’ anotherGeometry.
* Within : Returns 1 (TRUE) if this geometric object is “spatially within” anotherGeometry.
* Contains : Returns 1 (TRUE) if this geometric object “spatially contains” anotherGeometry.
* Overlaps : Returns 1 (TRUE) if this geometric object “spatially overlaps” anotherGeometry.
* Relate : Returns 1 (TRUE) if this geometric object is spatially related to anotherGeometry by testing for intersections between the interior, boundary and exterior of the two geometric objects as specified by the values in the intersectionPatternMatrix.
* LocateAlong : Returns a derived geometry collection value that matches the specified m coordinate value
* LocateBetween : Returns a derived geometry collection value that matches the specified range of m coordinate values inclusively.

**Comment by SergeStinckwich** [comment]
This is interesting, but this is a whole project and it will take us some time. If you want to go in that direction, we can open a separate repository for this.

What would be the scenario of using such operators with CORMAS?

**Comment by ElCep** [comment]
Above all, I opened these issues to keep these points in mind (and how to formulate them). I don't know if this is a completely different project. It seems to me that if we want to use spatial data, we should have space operators. 

regarding scenario , in cormas a shape (e.g. a plot) would be able to update its biomass when it contains a certain number of agents.

Space can be a Pandora's box. :alien:
